hpx/runtime/components/default_distribution_policy.hpp

```diff
@@ -105,7 +107,8 @@ namespace hpx { namespace components
         {
             using components::stub_base;
 
-            for (hpx::id_type const& loc: localities_)
+            auto localities = ensure_localities();
```

Comment:
taking a copy here seems to be a little overkill here.

Reply:
This does not take a copy of the vector, just a copy of the shared_ptr - no harm.

Reply:
unnecessary reference count.

Reply:
Come on - really?

Reply:
Yeah, shared ownership isn't needed in this particular case, so why artificially introduce it here? Sure, it sounds like minor nitpicking... I just think we shouldn't artificially introduce unnecessary overheads when they are avoidable.

Reply:
I don't see how we could avoid this in this case. `ensure_localities()` ensures having an initialized vector of localities in a thread-safe manner.